### ISSUE TITLE
Assertion failed during Reshape Op

### ISSUE THREAD

**Issue by shizy818** [onnx/onnx-mlir#490]
It failed to compile this model:
```
Assertion failed: (isa<U>()), function cast, file /Users/shizy/github/llvm-project/mlir/include/mlir/IR/Types.h, line 241.
```

![图片](https://user-images.githubusercontent.com/2142335/105973626-f2036800-60c7-11eb-899e-505294406cca.png)

I think it is input `rnn_h` with shape [1, -1, 256] that cause the problem.

**Comment by shizy818** [comment]
onnx model:
[keras_rnn_stock.onnx.zip](https://github.com/onnx/onnx-mlir/files/5879285/keras_rnn_stock.onnx.zip)

**Comment by tungld** [comment]
@shizy818 thank you for pointing out the issue! Fixed it here #498.

**Comment by shizy818** [comment]
@tungld Got it, thanks!
